repose-aggregator/artifacts/valve/src/main/scala/org/openrepose/valve/ReposeJettyServer.scala

```diff
@@ -126,37 +126,63 @@ class ReposeJettyServer(val nodeContext: AbstractApplicationContext,
 
         //Handle the Protocols and Ciphers
         //varargs are annoying, so lets deal with this using the scala methods
+        // From https://www.eclipse.org/jetty/documentation/9.4.x/configuring-ssl.html
+        // "When working with Includes / Excludes, it is important to know that Excludes will always win."
+        // TODO: The SslContextFactory's set Include/Exclude Protocols/CipherSuites override the built-in defaults.
+        // TODO: We may want to change at least the Excludes back to add's for v9.x.
         import scala.collection.JavaConversions._
-        Option(ssl.getExcludedProtocols).foreach(xp => cf.addExcludeProtocols(xp.getProtocol.toList: _*))
         Option(ssl.getIncludedProtocols).foreach { ip =>
           if (ip.getProtocol.nonEmpty) {
             cf.setIncludeProtocols(ip.getProtocol.toList: _*)
           }
-
         }
-        Option(ssl.getExcludedCiphers).foreach { xc =>
-          cf.addExcludeCipherSuites(xc.getCipher.toList: _*)
+        Option(ssl.getExcludedProtocols).foreach { ep =>
+          if (ep.getProtocol.nonEmpty) {
+            cf.setExcludeProtocols(ep.getProtocol.toList: _*)
+          }
         }
-
         Option(ssl.getIncludedCiphers).foreach { ic =>
           if (ic.getCipher.nonEmpty) {
             cf.setIncludeCipherSuites(ic.getCipher.toList: _*)
           }
         }
+        ////////////////////////////////////////////////////////////////////////////////
+        // TODO: Do not clear the Jetty defaults in v9.x
+        cf.setExcludeCipherSuites()
+        ////////////////////////////////////////////////////////////////////////////////
+        Option(ssl.getExcludedCiphers).foreach { ec =>
+          if (ec.getCipher.nonEmpty) {
+            cf.setExcludeCipherSuites(ec.getCipher.toList: _*)
+          }
+        }
+        ////////////////////////////////////////////////////////////////////////////////
+        // TODO: Do not add the old Jetty defaults in v9.x
+        // These were the originals that were changed in Jetty at the following commit:
+        // https://github.com/eclipse/jetty.project/commit/0a1b0b2bc69ea7e7f5f44992f47a84f926cdeebb
+        cf.addExcludeCipherSuites(
+          List(
+            "SSL_RSA_WITH_DES_CBC_SHA",
+            "SSL_DHE_RSA_WITH_DES_CBC_SHA",
+            "SSL_DHE_DSS_WITH_DES_CBC_SHA",
+            "SSL_RSA_EXPORT_WITH_RC4_40_MD5",
+            "SSL_RSA_EXPORT_WITH_DES40_CBC_SHA",
+            "SSL_DHE_RSA_EXPORT_WITH_DES40_CBC_SHA",
+            "SSL_DHE_DSS_EXPORT_WITH_DES40_CBC_SHA"): _*)
```

Comment:
Were these defaults present in Jetty 9.2.0 (currently in-use by the `master` branch), or did they come in on of the other upgrades? If the latter, do we actually need to add them back in?

Reply:
Looking at the Jetty 9.2.0 source I've pulled down locally, I do not see the defaults referenced in your link. I wonder if that's why we were using `add` instead of `set`? They were functionally equivalent in Jetty 9.2.0?

Reply:
This is the commit they changed in:
* https://github.com/eclipse/jetty.project/commit/0a1b0b2bc69ea7e7f5f44992f47a84f926cdeebb

Reply:
Sure, I saw that in your comment. So it looks like the oldest tag that commit is in is a Jetty 9.3.7 tag. Which means the behavior from Jetty 9.2.0 (our current version) to Jetty 9.3.7 changed. To be more exact, Jetty 9.3.0 seems to have introduced the default excluded list, and Jetty 9.3.7 modified it. However, I believe we want to maintain current behavior (from Jetty 9.2.0) which means no default exclusions.

Reply:
Removed.